### ISSUE TITLE
Add support for HLS (m3u8 streams)

### ISSUE THREAD

**Issue by smsisko** [Automattic/pocket-casts-android#670]
### What problem does this address?

Some podcasts are published using the HTTP Live Streaming protocol.

Those streams can currently be used in Pocket Cast for iOS, but it's not working on Android.

### What is your proposed solution?

Implementing support of the protocol would remove the feature disparity.

### Did you search for existing feature suggestions?

- [X] I have searched for existing feature suggestions.

**Comment by smsisko** [comment]
So I was looking a little through the source code, and I saw that Pocket Casts seems to be using ExoPlayer for media playing, and HLS streams are supported using ExoPlayer. I tested using the ExoPlayer Stream Test, and it could stream one such streams without any problems.

**Comment by CookieyedCodes** [comment]
Maby it just needs to be checkmarked on then or the player needs to be updated 🤔🤔

**Comment by smsisko** [comment]
Yeah, I've been trying to run the debug build to see where the limitations come from, but I'm running into other problems (most HTTP queries end up being 502 errors, I'm not sure what I'm doing wrong).

**Comment by smsisko** [comment]
All right, I found the 502 problems were related to running the debug build instead of the debugProd build.

I was able to make the necessary changes locally to be able to use a HLS stream, with those changes the behavior is equivalent to the iOS Pocket Casts client.

It might be a good idea to block downloads for such streams, right now I can download, but that only downloads the playlist, and it can play from the local playlist, so I have to remove the download to be able to play again. But I'm not sure if that's a little out of scope.
